molgenis-data-annotators/src/main/java/org/molgenis/data/annotator/tabix/TabixReader.java

```diff
@@ -369,38 +364,38 @@ public String next() throws IOException
 					if (i >= 0) assert (curr_off == off[i].v); // otherwise bug
 					if (i < 0 || off[i].v != off[i + 1].u)
 					{ // not adjacent chunks; then seek
-						mFp.seek(off[i + 1].u);
-						curr_off = mFp.getFilePointer();
-						++n_seeks;
+						blockCompressedInputStream.seek(off[i + 1].u);
+						curr_off = blockCompressedInputStream.getFilePointer();
 					}
 					++i;
 				}
 				String s;
-				if ((s = readLine(mFp)) != null)
+				if ((s = readLine(blockCompressedInputStream)) != null)
 				{
 					TIntv intv;
 					char[] str = s.toCharArray();
-					curr_off = mFp.getFilePointer();
+					curr_off = blockCompressedInputStream.getFilePointer();
 					if (str.length == 0 || str[0] == mMeta) continue;
 					intv = getIntv(s);
-					if (intv.tid != tid || intv.beg >= end) break; // no need to proceed
-					else if (intv.end > beg && intv.beg < end) return s; // overlap; return
+					if (intv.internalChromosomeID != internalChromosomeID || intv.beginPosition >= endPosition) break; // no need to proceed
+					else if (intv.end > beginPosition && intv.beginPosition < endPosition) return s; // overlap; return
 				}
 				else break; // end of file
 			}
-			iseof = true;
+			isEndOfFile = true;
+			blockCompressedInputStream.close();
```

Comment:
There's no guarantee that the iterator will actually be fully read right up to the end.
Until it is, the newly created stream will remain open.